src/main/java/at/jugger/tracker/service/impl/UserServiceImpl.java

```diff
@@ -30,7 +30,9 @@ public User getUser(@NotNull Long id) {
 
     @Override
     public User getUserByEmail(@NotNull String email) {
-        return userMapper.toDto(userRepository.findByEmail(email));
+        User user = userMapper.toDto(userRepository.findByEmail(email));
+        user.setEmail(email);
```

Comment:
Maybe explain why this call is here. This seems pretty weird without our knowledge